kf_lib_data_ingest/common/file_retriever.py

```diff
@@ -182,18 +216,31 @@ def get(self, url, auth=None):
         # I think there may be a better way to handle exceptional cleanup. -Avi
         try:
             if url not in self._files:
-                # Temporary file object that optionally deletes itself at the
-                # end of FileRetriever instance scope (if the enclosing folder
-                # isn't already being deleted)
-                self._files[url] = tempfile.NamedTemporaryFile(
-                    dir=self.storage_dir,
-                    delete=self.cleanup_at_exit and not self.__tmpdir
-                )
-                # Fetch the remote data
-                FileRetriever._getters[protocol](
-                    protocol, path, self._files[url],
-                    auth=auth, logger=self.logger
-                )
+                # Settings for temporary file object that optionally deletes
+                # itself at the end of FileRetriever instance scope
+                # (if the enclosing folder isn't already being deleted)
+                delete = self.cleanup_at_exit and not self.__tmpdir
+
+                # Invoke http getter
+                # Note - Temp file must be created inside _web_save since we
+                # won't know file ext until after file is requested
+                if (protocol == 'http') or (protocol == 'https'):
```

Comment:
I tried to see if there was a way to set the NamedTemporaryFile's suffix after it's instantiated, but there doesn't seem to be a way unfortunately